# Data Collection
League of Legends is a game created by the company known as 'Riot Games'. Riot Games has developed an API that is available through the Riot Games Developer Portal. This portal can be accessed by anyone with a Riot account. The API has methods that request information about different aspects of their game such as the people who play it and the games that are played.

We used a package called RiotWatcher, which has implementation for all public methods available in the Riot API, specificially for python.

League of Legends has a ranked system which places players into the 'appropriate' rank/division based on wins/losses and game performance.
The aim is to use the specialised methods to gather multiple databases of game information which will represent the different ranks.

### Imported Libraries

In [9]:
from os import listdir
import os
import sys
import csv
import pandas as pd

# installs the riotwatcher package to your machine
!{sys.executable} -m pip install riotwatcher

In [5]:
from riotwatcher import LolWatcher, ApiError

#API code
lol_watcher = LolWatcher('RGAPI-6a7afe56-0fba-446b-8835-12fa43b8eeb2')
my_region = 'oc1'
shard = 'AMERICAS'
limit = 12

lol_watcher stores and uses the API code provided at the time of running the program. This code expires 24 hours after being generated. The code tells the Riot who is accessing their functions and generally should not be shared. In the case of misuse Riot may blacklist the account/API code.
The code is manually generated through the developer portal.

In [3]:
#player info format for reference
summoner = 'mephistopheles1'
player = lol_watcher.summoner.by_name(my_region, summoner)
league = lol_watcher.league.by_summoner(my_region, player.get('id'))
for key in player:
    print(key,':',player[key])
print('\n')
for i in range(len(league)):
    for key in league[i]:
        print(key,':',league[i][key])

id : MgQEYLSiVaKohzmjNfinHVKHFAyRg4FCEN2EQGz8cHgsag
accountId : JYHTOV1zLbRs5QKOTOYvHP2LDDzMA5Pr1mgx-m9yROaHuwY
puuid : yXs9BSQz13E8Q-c5Sw6shdw7MVlt2zZJGTBh-3ew1IX4FxpUN6QoZrxxVUIVEkj-B9HMPUZ3yD60fA
name : Mephistopheles1
profileIconId : 774
revisionDate : 1635846594000
summonerLevel : 69


leagueId : 6c9f54d5-f1d2-47fa-a53c-0dfe28049ae8
queueType : RANKED_FLEX_SR
tier : GOLD
rank : IV
summonerId : MgQEYLSiVaKohzmjNfinHVKHFAyRg4FCEN2EQGz8cHgsag
summonerName : Mephistopheles1
leaguePoints : 9
wins : 32
losses : 36
veteran : False
inactive : False
freshBlood : False
hotStreak : False
leagueId : dd57d57f-42b0-4b1c-a080-e84d833e108d
queueType : RANKED_SOLO_5x5
tier : GOLD
rank : III
summonerId : MgQEYLSiVaKohzmjNfinHVKHFAyRg4FCEN2EQGz8cHgsag
summonerName : Mephistopheles1
leaguePoints : 0
wins : 84
losses : 84
veteran : False
inactive : False
freshBlood : False
hotStreak : False


In [4]:
#ranked match detail format for reference (only shows info for 1 out of 10 players that are provided)
#intially stored as a nested dictionary 
summoner = 'mephistopheles1'
player = lol_watcher.summoner.by_name(my_region, summoner)
matchList = lol_watcher.match_v5.matchlist_by_puuid('AMERICAS', player.get("puuid"))
for matchID in matchList:
    match = lol_watcher.match_v5.by_id('AMERICAS', matchID)
    if(match.get("info").get("queueId") == 420):
        for key in match:
            print("\n",key,":")
            for key2 in match[key]:
                if (key=='info') and (key2=='participants'):
                    for key3 in match['info']['participants'][0]:
                            print(key3,":",match[key][key2][0][key3])
                else:
                    print(key2,":",match[key][key2])
        break


 metadata :
dataVersion : 2
matchId : OC1_490592460
participants : ['SXPoxSsfSppEylaqdoGHuUQVKIwJjo6Ozd7mmagWbcd-mPrxUg04i0EAFMdQrz3B1nGbvVsM4WCSaw', 'LTay_R-uNwRyN_myflf6jY-u5V7-c04utAKg49OnnaAWeDCUg8ppXkt4KRbR99_6T-2c-jdfmFmyNQ', 'aMqyu2seca--Etga3B8OTy8j1vY4u4UxVD-sYmiPdMggmSBs2tDaTMBvkWCiCfQTkqFyZ_xn9dRQlg', '8z26u9Y7t3uoW0y0pg7YVn6NVdGb4PgkOq4Q6qqCz43N-HwNBeqK4vNZzv_AgDLPMurtNPMrtYmmlA', 'PhwNq_6ixG0Cg9pkhshC66miQ-YUZqsFwLGp9vqETKn9WPiu_IF00IHzjcclRyaIaU6Dbgh0o37yeQ', 'SGNM6WBkn0P-D7AVC5lt0AdJMrwLwRDPX50pq8rKm_oGojVuly3Jpno3B_X8TJ3eNl5ZXZ-r4KQV7w', 'lKrcjib4tobivTjEYPz6h58Bj8s41RVQGznCbBHSOeunGshI3v460mxyvNAyzDZORAax6nPpZXRqCA', 'jtY1gTQDUemqtJaj5Nal5dqZbDbR13g4y06njOgWhm7syzv1jEeYClAN7SqrKAjfR_7ffJcUQHjMpA', 'yXs9BSQz13E8Q-c5Sw6shdw7MVlt2zZJGTBh-3ew1IX4FxpUN6QoZrxxVUIVEkj-B9HMPUZ3yD60fA', '170bObneRuYzu8XCBZhADSxOxcABZI_ocr8I46GVYhCa-Y9UbquL-cOX8fFAhaDjqPB0Zj_xYobqIA']

 info :
gameCreation : 1635844368000
gameDuration : 2047
gameEndTimestamp : 1635846592266
gameId : 490592460
g

In [6]:
#get the ranked entries for the bottom of each division in each tier
divisions=['I','II','III','IV']
tiers=['IRON','BRONZE','SILVER','GOLD','PLATINUM','DIAMOND']
entries=[]
for tier in tiers:
    for division in divisions:
        entries.append(lol_watcher.league.entries(my_region,'RANKED_SOLO_5x5',tier,division))

In [7]:
count = 0
for tier in tiers:
    for division in divisions:
        print(tier + " " + division +": " + str(len(entries[count])) + " entries")
        count += 1

IRON I: 205 entries
IRON II: 205 entries
IRON III: 205 entries
IRON IV: 205 entries
BRONZE I: 205 entries
BRONZE II: 205 entries
BRONZE III: 205 entries
BRONZE IV: 205 entries
SILVER I: 205 entries
SILVER II: 205 entries
SILVER III: 205 entries
SILVER IV: 205 entries
GOLD I: 205 entries
GOLD II: 205 entries
GOLD III: 205 entries
GOLD IV: 205 entries
PLATINUM I: 205 entries
PLATINUM II: 205 entries
PLATINUM III: 205 entries
PLATINUM IV: 205 entries
DIAMOND I: 205 entries
DIAMOND II: 205 entries
DIAMOND III: 205 entries
DIAMOND IV: 205 entries


There are **6 overall ranks** ('IRON', 'BRONZE', 'SILVER', 'GOLD', 'PLATINUM', 'DIAMOND'). In each rank there are **4 divisions** which even further splits up the players.
Using the **.league.entries** method we are given the **bottom 205 players for each rank/division**.

In [7]:
#limiting the number of entries per division from 205 to 12 and combining each rank's entries (48 entries per rank)
iron = []
bronze = []
silver = []
gold = []
platinum = []
diamond = []

#count is keeping track of rank/division, counter is keeping track of number of entries per rank
counter = 0
for count, entry in enumerate(entries):
    for summoner in entry:
        if(count < 4 and counter < limit):
            iron.append(summoner)
            counter += 1
            if(counter == limit):
                counter = 0
                break
        elif(count < 8 and counter < limit):
            bronze.append(summoner)
            counter += 1
            if(counter == limit):
                counter = 0
                break
        elif(count < 12 and counter < limit):
            silver.append(summoner)
            counter += 1
            if(counter == limit):
                counter = 0
                break
        elif(count < 16 and counter < limit):
            gold.append(summoner)
            counter += 1
            if(counter == limit):
                counter = 0
                break
        elif(count < 20 and counter < limit):
            platinum.append(summoner)
            counter += 1
            if(counter == limit):
                counter = 0
                break
        elif(count < 24 and counter < limit):
            diamond.append(summoner)
            counter += 1
            if(counter == limit):
                counter = 0
                break
        
rank = {
    "iron": iron,
    "bronze": bronze,
    "silver": silver,
    "gold": gold,
    "platinum": platinum,
    "diamond": diamond
}

Since each database represents a rank, we need to combine the divisions within each rank. The above code performs this task while also **dropping the number of entries included from 205 to 12** due to the excessive size of the databases that would come from 820 (4*205) players per rank.

In [8]:
#getting puuid from player entries
for r in rank:
    for count, player in enumerate(rank[r]):
        counter = count%48
        try:
            rank[r][counter] = lol_watcher.summoner.by_name(my_region,rank[r][counter]['summonerName'])
        except:
            print(rank[r].pop(counter))
            continue
    print("FINISHED: " + r)

FINISHED: iron
FINISHED: bronze
FINISHED: silver
{'leagueId': '93a78579-1057-4438-9791-c64fadad3406', 'queueType': 'RANKED_SOLO_5x5', 'tier': 'GOLD', 'rank': 'IV', 'summonerId': 'dfwOtYmd7LuyGjHiKrvg-21N0u4fbvnZnqsuTGpTfG9Pr2I', 'summonerName': 'Utahime', 'leaguePoints': 39, 'wins': 98, 'losses': 114, 'veteran': False, 'inactive': False, 'freshBlood': False, 'hotStreak': False}
FINISHED: gold
FINISHED: platinum
FINISHED: diamond


In order to get the match history of each player we have to get the player's **puuid** which can be found by using the **.summoner.by_name** method on each of the entries.

In [9]:
#number of players in each rank
for r in rank:
    print(r + " players: " + str(len(rank[r])))

iron players: 48
bronze players: 48
silver players: 48
gold players: 47
platinum players: 48
diamond players: 48


In [10]:
#getting the last 20 matchIds for each player using their puuid
iron = []
bronze = []
silver = []
gold = []
platinum = []
diamond = []

matches = {
    "iron": iron,
    "bronze": bronze,
    "silver": silver,
    "gold": gold,
    "platinum": platinum,
    "diamond": diamond
}

for r in rank:
    for count, player in enumerate(rank[r]):
        counter = count%48
        try:
            temp=lol_watcher.match_v5.matchlist_by_puuid(shard,rank[r][counter]['puuid'])
            for match in temp:
                matches[r].append(match)
        except:
            continue

Using a method called **.match_v5.matchlist_by_puuid** we get the last 20 games of the player whose **puuid** we have specified.
The 20 games are in the form of IDs which are attached to a dictionary that we have initialised.

In [11]:
#fetching the match information from matchIds and filtering for ranked (has very long runtime)
iron = []
bronze = []
silver = []
gold = []
platinum = []
diamond = []

ranked = {
    "iron": iron,
    "bronze": bronze,
    "silver": silver,
    "gold": gold,
    "platinum": platinum,
    "diamond": diamond
}

for r in matches:
    for match in matches[r]:
        try:
            game = lol_watcher.match_v5.by_id(shard,match)
            if(game['info']['queueId'] == 420):
                ranked[r].append(game)
        except:
            continue
    print("FINISHED " + r)

FINISHED iron
FINISHED bronze
FINISHED silver
FINISHED gold
FINISHED platinum
FINISHED diamond


The **.match_v5.by_id** gets the game information using each games ID number. The format of the output has been demonstrated earlier in the code. We also filter the games by whether or not they are ranked (**queueId == 420**) since there are other game modes in the game.

In [21]:
#number of ranked matches appended to each rank
for r in ranked:
    print(r + ": " + str(len(ranked[r])))

iron: 334
bronze: 402
silver: 472
gold: 416
platinum: 450
diamond: 492


Since not every game played in a player's last 20 games are ranked, there is inconsistency in database sizes between ranks.

It is observed that, higher the rank, greater the proportion of games from their last 20 matches are ranked (iron 334, diamond 492). This may imply that iron players care less about ranked and enjoy/focus on other aspects of the game more.

## Generating the Databases
We are creating a csv file for each rank. The fields will be found through filtering the match information (nested dictionary) for what we need.

In [46]:
#creating new csv file for each rank
for r in ranked:
    filename = 'Data/'+ r + '.csv'
    with open(filename, 'w', newline = '') as file:
        writer = csv.writer(file)
        writer.writerow(['matchId','gameDuration','visionScore','wardsBought','consumablesPurchased','damageDealtToChampions',
                         'totalHealing','timeSpentDead','ccTimeDealt','minionKills','totalGold','championLevel','summonerLevel',
                         'spell1Casts','spell2Casts','spell3Casts','spell4Casts','summoner1Casts','summoner2Casts',
                         'firstBaron','baronKills','firstDragon','dragonKills','firstInhibitor','inhibitorKills',
                         'firstTower','towerKills','damageToObjectives','firstBlood','championKills',
                         'firstRiftHerald','riftHeraldKills',
                         'champion1','champion2','champion3','champion4','champion5',
                         'enemyChampion1','enemyChampion2','enemyChampion3','enemyChampion4','enemyChampion5',
                         '1ban1','1ban2','1ban3','1ban4','1ban5','2ban1','2ban2','2ban3','2ban4','2ban5', 'teamId', 'win'])

In [47]:
#each game shows information for the 10 players that participate (5 on winning side, 5 on losing side)
#filtering game information to create a record for each team 
for r in ranked:
    filename = 'Data/'+ r + '.csv'
    with open(filename, 'a', newline = '') as file:
        for match in ranked[r]:
            
            #duration is inconsistently written in seconds or milliseconds 
            duration = match['info']['gameDuration']
            if(len(str(duration)) > 5):
                duration = int(duration/1000)
            
            #initialising variables that need to be accumulated from each team's participants
            winVisionScore = 0
            winWardsBought = 0
            winConsumablesBought = 0
            winDamageDealtToChampions = 0
            winHealing = 0
            winTimeDead = 0
            winCCTimeDealt = 0
            winMinionKills = 0
            winTotalGold = 0
            winChampionLevel = 0
            winSummonerLevel = 0
            winSpell1Cast = 0
            winSpell2Cast = 0
            winSpell3Cast = 0
            winSpell4Cast = 0
            winSummoner1Casts = 0
            winSummoner2Casts = 0
            winDamageToObjectives = 0
            winChamps = []
            
            loseVisionScore = 0
            loseWardsBought = 0
            loseConsumablesBought = 0
            loseDamageDealtToChampions = 0
            loseHealing = 0
            loseTimeDead = 0
            loseCCTimeDealt = 0
            loseMinionKills = 0
            loseTotalGold = 0
            loseChampionLevel = 0
            loseSummonerLevel = 0
            loseSpell1Cast = 0
            loseSpell2Cast = 0
            loseSpell3Cast = 0
            loseSpell4Cast = 0
            loseSummoner1Casts = 0
            loseSummoner2Casts = 0
            loseDamageToObjectives = 0
            loseChamps = []
         
            #accumulating winning and losing team stats
            for participant in match['info']['participants']:
                if(participant['win'] == True):
                    winVisionScore += participant['visionScore']
                    winWardsBought += participant['visionWardsBoughtInGame']
                    winConsumablesBought += participant['consumablesPurchased']
                    winDamageDealtToChampions += participant['totalDamageDealtToChampions']
                    winHealing += participant['totalHeal']
                    winTimeDead += participant['totalTimeSpentDead']
                    winCCTimeDealt += participant['totalTimeCCDealt']
                    winMinionKills += participant['totalMinionsKilled']
                    winTotalGold += participant['goldEarned']
                    winChampionLevel += participant['champLevel']
                    winSummonerLevel += participant['summonerLevel']
                    winSpell1Cast += participant['spell1Casts']
                    winSpell2Cast += participant['spell2Casts']
                    winSpell3Cast += participant['spell3Casts']
                    winSpell4Cast += participant['spell4Casts']
                    winSummoner1Casts += participant['summoner1Casts']
                    winSummoner2Casts += participant['summoner2Casts']
                    winDamageToObjectives += participant['damageDealtToObjectives']
                    winChamps.append(participant['championId'])


                else:
                    loseVisionScore += participant['visionScore']
                    loseWardsBought += participant['visionWardsBoughtInGame']
                    loseConsumablesBought += participant['consumablesPurchased']
                    loseDamageDealtToChampions += participant['totalDamageDealtToChampions']
                    loseHealing += participant['totalHeal']
                    loseTimeDead += participant['totalTimeSpentDead']
                    loseCCTimeDealt += participant['totalTimeCCDealt']
                    loseMinionKills += participant['totalMinionsKilled']
                    loseTotalGold += participant['goldEarned']
                    loseChampionLevel += participant['champLevel']
                    loseSummonerLevel += participant['summonerLevel']
                    loseSpell1Cast += participant['spell1Casts']
                    loseSpell2Cast += participant['spell2Casts']
                    loseSpell3Cast += participant['spell3Casts']
                    loseSpell4Cast += participant['spell4Casts']
                    loseSummoner1Casts += participant['summoner1Casts']
                    loseSummoner2Casts += participant['summoner2Casts']
                    loseDamageToObjectives += participant['damageDealtToObjectives']
                    loseChamps.append(participant['championId'])

                    
            for team in match['info']['teams']:
                if(team['win'] == True):
                    try:
                        winTeamId = team['teamId']
                        winFirstBaron = team['objectives']['baron']['first']
                        winBaronKills = team['objectives']['baron']['kills']        
                        winFirstDragon = team['objectives']['dragon']['first']
                        winDragonKills = team['objectives']['dragon']['kills']
                        winFirstInhibitor = team['objectives']['inhibitor']['first']
                        winInhibitorKills = team['objectives']['inhibitor']['kills']
                        winFirstTower = team['objectives']['tower']['first']
                        winTowerKills = team['objectives']['tower']['kills']
                        winFirstBlood = team['objectives']['champion']['first']
                        winChampionKills = team['objectives']['champion']['kills']
                        winFirstRiftHerald = team['objectives']['riftHerald']['first']
                        winRiftHeraldKills = team['objectives']['riftHerald']['kills']
                        win = team['win']
                        winBan1 = team['bans'][0]['championId']
                        winBan2 = team['bans'][1]['championId']
                        winBan3 = team['bans'][2]['championId']
                        winBan4 = team['bans'][3]['championId']
                        winBan5 = team['bans'][4]['championId']
                    except:
                        continue

                        
                else:
                    try:
                        loseTeamId = team['teamId']
                        loseFirstBaron = team['objectives']['baron']['first']
                        loseBaronKills = team['objectives']['baron']['kills']        
                        loseFirstDragon = team['objectives']['dragon']['first']
                        loseDragonKills = team['objectives']['dragon']['kills']
                        loseFirstInhibitor = team['objectives']['inhibitor']['first']
                        loseInhibitorKills = team['objectives']['inhibitor']['kills']
                        loseFirstTower = team['objectives']['tower']['first']
                        loseTowerKills = team['objectives']['tower']['kills']
                        loseFirstBlood = team['objectives']['champion']['first']
                        loseChampionKills = team['objectives']['champion']['kills']
                        loseFirstRiftHerald = team['objectives']['riftHerald']['first']
                        loseRiftHeraldKills = team['objectives']['riftHerald']['kills']
                        lose = team['win']
                        loseBan1 = team['bans'][0]['championId']
                        loseBan2 = team['bans'][1]['championId']
                        loseBan3 = team['bans'][2]['championId']
                        loseBan4 = team['bans'][3]['championId']
                        loseBan5 = team['bans'][4]['championId']
                    except:
                        continue


            writer = csv.writer(file)
            
            #winning side record
            writer.writerow([match['metadata']['matchId'],
                            duration,
                            winVisionScore,
                            winWardsBought,
                            winConsumablesBought,
                            winDamageDealtToChampions,
                            winHealing,
                            winTimeDead,
                            winCCTimeDealt,
                            winMinionKills,
                            winTotalGold,
                            winChampionLevel,
                            winSummonerLevel,
                            winSpell1Cast, winSpell2Cast, winSpell3Cast, winSpell4Cast, 
                            winSummoner1Casts, winSummoner2Casts,
                            winFirstBaron, winBaronKills, 
                            winFirstDragon, winDragonKills,
                            winFirstInhibitor, winInhibitorKills,
                            winFirstTower, winTowerKills,
                            winDamageToObjectives,
                            winFirstBlood, winChampionKills,
                            winFirstRiftHerald, winRiftHeraldKills,
                            winChamps[0],winChamps[1],winChamps[2],winChamps[3],winChamps[4],
                            loseChamps[0],loseChamps[1],loseChamps[2],loseChamps[3],loseChamps[4],
                            winBan1,winBan2,winBan3,winBan4,winBan5,
                            loseBan1,loseBan2, loseBan3,loseBan4,loseBan5, winTeamId, win])
 
            #losing side record
            writer.writerow([match['metadata']['matchId'],
                            duration,
                            loseVisionScore,
                            loseWardsBought,
                            loseConsumablesBought,
                            loseDamageDealtToChampions,
                            loseHealing,
                            loseTimeDead,
                            loseCCTimeDealt,
                            loseMinionKills,
                            loseTotalGold,
                            loseChampionLevel,
                            loseSummonerLevel,
                            loseSpell1Cast, loseSpell2Cast, loseSpell3Cast, loseSpell4Cast, 
                            loseSummoner1Casts, loseSummoner2Casts,
                            loseFirstBaron, loseBaronKills, 
                            loseFirstDragon, loseDragonKills,
                            loseFirstInhibitor, loseInhibitorKills,
                            loseFirstTower, loseTowerKills,
                            loseDamageToObjectives,
                            loseFirstBlood, loseChampionKills,
                            loseFirstRiftHerald, loseRiftHeraldKills,
                            loseChamps[0],loseChamps[1],loseChamps[2],loseChamps[3],loseChamps[4],
                            winChamps[0],winChamps[1],winChamps[2],winChamps[3],winChamps[4],
                            winBan1,winBan2,winBan3,winBan4,winBan5,
                            loseBan1,loseBan2, loseBan3,loseBan4,loseBan5, loseTeamId, lose])

Since the game information shows information about the performance of all 10 individual players in the game, we have created two records per match which has the accumulated values for each field for each team. We felt the need to do this so our models could have a target (win = true or false).

In [10]:
#the generated data
filename = 'Data/iron.csv'
table = pd.read_csv(filename)
table.head()

matchId  gameDuration  visionScore  wardsBought  \
0  OC1_393708356          1852           56            1   
1  OC1_393708356          1852          105            3   
2  OC1_393714943          1854          121            4   
3  OC1_393714943          1854           67            3   
4  OC1_393710869          1448           63            3   

   consumablesPurchased  damageDealtToChampions  totalHealing  timeSpentDead  \
0                    10                   95606         36654            878   
1                     5                   74843         23935           1178   
2                    16                  102117         34586            968   
3                    21                   74015         27955           1270   
4                     8                   51209         23131            383   

   ccTimeDealt  minionKills  ...  1ban3  1ban4  1ban5  2ban1  2ban2  2ban3  \
0          824          518  ...     99    238     84    875     63    266   
1         1233          423  ...     99    238     84    875     63    266   
2         1336          626  ...    360      2    350    360    143    238   
3         1397          502  ...    360      2    350    360    143    238   
4         2300          414  ...    238     63     36     89     45     54   

   2ban4  2ban5  teamId    win  
0    157    111     200   True  
1    157    111     100  False  
2     84    526     200   True  
3     84    526     100  False  
4    122      7     100   True  

[5 rows x 54 columns]

In [11]:
#list directory of csv files in data folder
CSV_Files = [file for file in listdir('Data/') if file.endswith('.csv')]
print(CSV_Files)

['bronze.csv', 'diamond.csv', 'gold.csv', 'iron.csv', 'platinum.csv', 'silver.csv']


## Functions for Reading Libraries 
The libraries read, have values stored that are related to different ID's for fields that are or were in the databases. For example 1ban3 = 99 is the 3rd player from team 1's ban, which in this case was 'Lux'.

In [4]:
import json

def printLibrary(dic,dict_keys,list_keys=[]):
    for key in dic.keys():
        if key in dict_keys:
            print("{0}: {1}".format(key,list(dic.get(key).keys())))
        elif key in list_keys:
            print("{0}: {1}".format(key,dic.get(key)[0]))
        else:
            print("{0}: {1}".format(key,dic.get(key)))

In [5]:
with open('Libraries/champion.json', encoding="utf8") as f:
    champion = json.load(f)

printLibrary(champion,['data'])

type: champion
format: standAloneComplex
version: 11.20.1
data: ['Aatrox', 'Ahri', 'Akali', 'Akshan', 'Alistar', 'Amumu', 'Anivia', 'Annie', 'Aphelios', 'Ashe', 'AurelionSol', 'Azir', 'Bard', 'Blitzcrank', 'Brand', 'Braum', 'Caitlyn', 'Camille', 'Cassiopeia', 'Chogath', 'Corki', 'Darius', 'Diana', 'Draven', 'DrMundo', 'Ekko', 'Elise', 'Evelynn', 'Ezreal', 'Fiddlesticks', 'Fiora', 'Fizz', 'Galio', 'Gangplank', 'Garen', 'Gnar', 'Gragas', 'Graves', 'Gwen', 'Hecarim', 'Heimerdinger', 'Illaoi', 'Irelia', 'Ivern', 'Janna', 'JarvanIV', 'Jax', 'Jayce', 'Jhin', 'Jinx', 'Kaisa', 'Kalista', 'Karma', 'Karthus', 'Kassadin', 'Katarina', 'Kayle', 'Kayn', 'Kennen', 'Khazix', 'Kindred', 'Kled', 'KogMaw', 'Leblanc', 'LeeSin', 'Leona', 'Lillia', 'Lissandra', 'Lucian', 'Lulu', 'Lux', 'Malphite', 'Malzahar', 'Maokai', 'MasterYi', 'MissFortune', 'MonkeyKing', 'Mordekaiser', 'Morgana', 'Nami', 'Nasus', 'Nautilus', 'Neeko', 'Nidalee', 'Nocturne', 'Nunu', 'Olaf', 'Orianna', 'Ornn', 'Pantheon', 'Poppy', 'Pyke',

In [6]:
with open('Libraries/item.json', encoding="utf8") as f:
    item = json.load(f)

printLibrary(item,['basic','data'],['groups','tree'])

type: item
version: 11.20.1
basic: ['name', 'rune', 'gold', 'group', 'description', 'colloq', 'plaintext', 'consumed', 'stacks', 'depth', 'consumeOnFull', 'from', 'into', 'specialRecipe', 'inStore', 'hideFromAll', 'requiredChampion', 'requiredAlly', 'stats', 'tags', 'maps']
data: ['1001', '1004', '1006', '1011', '1018', '1026', '1027', '1028', '1029', '1031', '1033', '1035', '1036', '1037', '1038', '1039', '1042', '1043', '1052', '1053', '1054', '1055', '1056', '1057', '1058', '1082', '1083', '2003', '2010', '2015', '2031', '2033', '2051', '2052', '2055', '2065', '2138', '2139', '2140', '2403', '2419', '2420', '2421', '2422', '2423', '2424', '3001', '3003', '3004', '3006', '3009', '3011', '3020', '3024', '3026', '3031', '3033', '3035', '3036', '3040', '3041', '3042', '3044', '3046', '3047', '3050', '3051', '3053', '3057', '3065', '3066', '3067', '3068', '3070', '3071', '3072', '3074', '3075', '3076', '3077', '3078', '3082', '3083', '3085', '3086', '3089', '3091', '3094', '3095', '3100'

In [7]:
with open('Libraries/summoner.json', encoding="utf8") as f:
    summoner = json.load(f)
    
printLibrary(summoner,['data'])

type: summoner
version: 11.20.1
data: ['SummonerBarrier', 'SummonerBoost', 'SummonerDot', 'SummonerExhaust', 'SummonerFlash', 'SummonerHaste', 'SummonerHeal', 'SummonerMana', 'SummonerPoroRecall', 'SummonerPoroThrow', 'SummonerSmite', 'SummonerSnowURFSnowball_Mark', 'SummonerSnowball', 'SummonerTeleport', 'Summoner_UltBook_Placeholder']
